src/types/integrations.d.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Integrations
+ */
+import { Identifiable, CrudQueryableResource, ResourceList } from './core'
+
+/**
+ * Core Integration Base Interface
+ */
+export interface IntegrationBase {
+  type: 'integration'
+  name: string
+  description: string
+  enabled: boolean
+  observes: string[]
+  integration_type: string
+  configuration: any
+}
+
+export interface Integration extends Identifiable, IntegrationBase {}
+
+export interface IntegrationJob extends Identifiable {
+  type: 'integration-job'
+}
+
+export interface IntegrationFilter {}
+
+/**
+ * Interface for an Integration contained within a Log
+ */
+export interface LogIntegration
+  extends Identifiable,
+    Omit<
```

Comment:
If we use the `Omit` utility to remove almost all the values from the original source type `Integration` then would we be better using the [Pick utility](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys)

Reply:
Integration does already extend Identifiable but if we change from Omit to Pick we need to make sure to include the properties we want to keep. e.g. id in this case.